src/entities/swap/swaps/v3/index.ts

```diff
@@ -131,17 +147,30 @@ export class SwapV3 implements SwapBase {
             };
         }
         if ('exactAmountOut' in this.swaps) {
-            const { result } =
-                await routerContract.simulate.querySwapSingleTokenExactOut(
-                    [
-                        this.swaps.pool,
-                        this.swaps.tokenIn,
-                        this.swaps.tokenOut,
-                        this.swaps.exactAmountOut,
-                        DEFAULT_USERDATA,
-                    ],
-                    { blockNumber: block },
-                );
+            const { result } = account
+                ? await routerContract.simulate.swapSingleTokenExactOut(
+                      [
+                          this.swaps.pool,
+                          this.swaps.tokenIn,
+                          this.swaps.tokenOut,
+                          this.swaps.exactAmountOut,
+                          999999999999999999n, // maxAmountIn
```

Comment:
Setting `maxAmoutnIn` like this is not ideal but not sure how to handle because `getLimitAmount` helper function requires  query results

Reply:
You can use `MAX_UINT256` instead.
Not quite following what the issue is here (particularly in relation to getLimitAmount). Can you elaborate?

Reply:
I noticed that `getLimitAmount()` is used to calculate the min/max amounts in/out, but it requires `queryOutput` which doesnt exist yet in the life cycle

But we are all good since you explained it's okay to just use `0n` and `MAX_UINT256`